xchange-poloniex/src/main/java/info/bitrich/xchangestream/poloniex/PoloniexStreamingMarketDataService.java

```diff
@@ -22,10 +24,27 @@
 import java.util.*;
 
 public class PoloniexStreamingMarketDataService implements StreamingMarketDataService {
+    public static final String TICKER_CHANNEL_NAME = "ticker";
     private final WampStreamingService streamingService;
+    private Supplier<Observable<Ticker>> streamingTickers;
 
     public PoloniexStreamingMarketDataService(WampStreamingService streamingService) {
         this.streamingService = streamingService;
+        this.streamingTickers = Suppliers.memoize(streamingService.subscribeChannel(TICKER_CHANNEL_NAME)
```

Comment:
Hm, this is fine but is making the (correct) assumption that `NettyStreamingService.subscribeChannel` doesn't actually do very much synchronously - the actual initialisation of the stream occurs when the `Observable` is subscribed, so we can afford to call `subscribeChannel` here without anything much actually happening as long as we memoise the call to `share()`.

This feels a bit brittle though, since `subscribeChannel` might change at some point to do a bit more in-line.  It's also easy to avoid - simply memoise the entire thing:

```
Suppliers.memoise(() -> streamingService.subscribeChannel()
   ...
   .share());
```

Reply:
Done.